### PR TITLE
Add metadata to results

### PR DESCRIPTION
**Pull request by buthter**
- Added metadata struct that keeps track of priority, functions, and categories for a single test
- Load a map of test names to TestMetadata from test_metadata.csv
- Include these metadata fields in the test output
- Refactor some related test result functions

### CODE REVIEW

**Comment by erademacher** [comment]
Accidentally logged into github with the wrong credentials... Oh well.

**Review by pvanderknyff — APPROVED** [review]
LGTM, though I would still wait for a review from Luke or Logan to make sure nothing unexpected will break in TDVT's guts.

**Comment by erademacher** [comment]
@lukewrites Would you mind taking a quick look to make sure I'm not doing anything obviously dumb here?

File: tdvt/tdvt/tdvt_core.py [hunk 1]
```diff
@@ -84,42 +97,41 @@ def add_test_result_error(self, test_file, result, is_first=True):
                 result.saved_error_message = "Error. Previous error message is: " + self.saved_error_message
         self.add_test_result(test_file, result)
 
-    def add_timeout_test_failure(self, test_result_file):
+    def create_test_result(self, test_result_file, test_category):
```
Comment (lukewrites) [inline tdvt/tdvt/tdvt_core.py:100]: This is a great change 👍

**Pull request merged**